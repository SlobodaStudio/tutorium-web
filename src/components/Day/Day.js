import React from 'react'
import { List, ListItem, ListSubHeader} from 'react-toolbox/lib/list'
import styles from './Day.scss'

function Day({ dateAndEvents: { date, events } }) {
  return (
    <ListItem key={date} className={"day-сontent"} >
      <List>
        <ListSubHeader caption={date} />
        {events.map((event) =>
          <ListItem key={event} caption={event} className={styles.event} />
        )}
      </List>
    </ListItem>
  );
};

export default Day
