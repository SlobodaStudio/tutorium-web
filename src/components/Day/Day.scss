@import '../../variables';

.event{
  border-bottom: 1px solid $light-grey;
}
