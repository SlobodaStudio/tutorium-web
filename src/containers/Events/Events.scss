@import '../../variables';

.container__list {
  margin: 50px auto;
  max-width: $tablet-width;
}
