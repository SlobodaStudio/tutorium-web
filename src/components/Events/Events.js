import React from 'react'
import { List} from 'react-toolbox/lib/list'
import Day from 'components/Day/Day'
import styles from './Events.scss'

function DaysList(props) {
  return (
    <List>
      {props.dates.map(date =>
        <Day dateAndEvents={date} />
      )}
    </List>
  );
};

class Events extends React.Component {
  constructor(props) {
    super(props);
    this.state = {dates: [
        {
          date: "March, 10",
          events: [
            "Machine learning practice"
          ]
        },
        {
          date: "April, 5",
          events: [
            "Ruby vs JavaScript: be quick or be dead",
            "Super Duper JavaScript"
          ]
        }
      ]
    };
  }
  render() {
    return (
      <div className={styles.container__list}>
        <h3>Upcoming Events</h3>
        <DaysList dates={this.state.dates} />
      </div>
    );
  }
}

export default Events
