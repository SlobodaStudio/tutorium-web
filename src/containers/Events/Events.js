import React from 'react'
import { List} from 'react-toolbox/lib/list'
import Day from '../../components/Day/Day'
import styles from './Events.scss'

class Events extends React.Component {
  constructor(props) {
    super(props);
    this.state = {dates: [
        {
          date: "March, 10",
          events: [
            "Machine learning practice"
          ]
        },
        {
          date: "April, 5",
          events: [
            "Ruby vs JavaScript: be quick or be dead",
            "Super Duper JavaScript"
          ]
        }
      ]
    };
  }
  render() {
    return (
      <div className={styles.container__list}>
        <h3>Upcoming Events</h3>
        <List>
          {this.state.dates.map(date =>
            <Day key={date.date} dateAndEvents={date} />
          )}
        </List>
      </div>
    );
  }
}

export default Events
