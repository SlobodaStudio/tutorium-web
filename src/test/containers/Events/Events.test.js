import React from 'react';
import ReactDOM from 'react-dom';
import { List, ListItem, ListSubHeader} from 'react-toolbox/lib/list'
import { expect } from 'chai';
import { shallow } from 'enzyme';

import Events from '../../../containers/Events/Events'
import Day from '../../../components/Day/Day'

const rootEl = document.createElement('div');

ReactDOM.render(
   <div>
      <Events />
   </div>,
  rootEl
);

describe('<Events />', () => {
  it('renders two <Day /> components', () => {
    const wrapper = shallow(<Events />);
    expect(wrapper.find(Day)).to.have.length(2);
  });

  it('renders `ListItem`s', () => {
    const wrapper = shallow(<Events />);
    expect(wrapper.find(ListItem)).exists;
  });
});
