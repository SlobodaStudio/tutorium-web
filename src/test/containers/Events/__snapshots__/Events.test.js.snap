exports[`test Events renders correctly 1`] = `
<div
  className={undefined}>
  <h3>
    Upcoming Events
  </h3>
  <ul
    className=""
    data-react-toolbox="list">
    <li
      className="undefined day-сontent"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}>
      <span
        className="">
        <span
          className={undefined}>
          <span
            className={undefined}
            onClick={undefined}
            onMouseDown={undefined}>
            <ul
              className=""
              data-react-toolbox="list">
              <h5
                className="">
                March, 10
              </h5>
              <li
                className="undefined "
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}>
                <span
                  className="">
                  <span
                    className="">
                    <span
                      className="undefined"
                      data-react-toolbox="list-item-text">
                      Machine learning practice
                    </span>
                  </span>
                </span>
              </li>
            </ul>
          </span>
        </span>
        <span
          className="" />
      </span>
    </li>
    <li
      className="undefined day-сontent"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}>
      <span
        className="">
        <span
          className={undefined}>
          <span
            className={undefined}
            onClick={undefined}
            onMouseDown={undefined}>
            <ul
              className=""
              data-react-toolbox="list">
              <h5
                className="">
                April, 5
              </h5>
              <li
                className="undefined "
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}>
                <span
                  className="">
                  <span
                    className="">
                    <span
                      className="undefined"
                      data-react-toolbox="list-item-text">
                      Ruby vs JavaScript: be quick or be dead
                    </span>
                  </span>
                </span>
              </li>
              <li
                className="undefined "
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}>
                <span
                  className="">
                  <span
                    className="">
                    <span
                      className="undefined"
                      data-react-toolbox="list-item-text">
                      Super Duper JavaScript
                    </span>
                  </span>
                </span>
              </li>
            </ul>
          </span>
        </span>
        <span
          className="" />
      </span>
    </li>
  </ul>
</div>
`;
